src/app/admin/manageUser/ResetPasswordForm.tsx

```diff
@@ -28,65 +28,67 @@ const ErrorMessage = styled.span<{ theme: DefaultTheme }>`
     color: ${(props) => props.theme.error};
 `;
 
-const ResetPasswordForm: React.FC<Props> = ({ userToEdit, onCancel, onConfirm }) => {
-    const [password, setPassword] = useState("");
-    const [errorMessage, setErrorMessage] = useState<string | null>(null);
+const updatePassword = async (
+    userId: string,
+    newPassword: string
+): Promise<UpdatePasswordResponse> => {
+    const response = await adminUpdateUserEmailAndPassword({
+        userId: userId,
+        attributes: { password: newPassword },
+    });
 
-    const updatePassword = async (
-        userId: string,
-        newPassword: string
-    ): Promise<UpdatePasswordResponse> => {
-        const response = await adminUpdateUserEmailAndPassword({
-            userId: userId,
-            attributes: { password: newPassword },
+    if (response.error) {
+        void logErrorReturnLogId(`Error resetting password userId: ${userId}`, {
+            response,
         });
+        return { errorMessage: response.error["Failed to update user"] };
+    }
+    return { errorMessage: null };
+};
 
-        if (response.error) {
-            void logErrorReturnLogId(`Error resetting password userId: ${userId}`, {
-                response,
-            });
-            return { errorMessage: response.error["Failed to update user"] };
-        }
-        return { errorMessage: null };
-    };
+const ResetPasswordForm: React.FC<Props> = ({ userToEdit, onCancel, onConfirm }) => {
+    const [password, setPassword] = useState("");
+    const [errorMessage, setErrorMessage] = useState<string | null>(null);
 
-    const onConfirmPassword = (): void => {
-        updatePassword(userToEdit.userId, password).then(({ errorMessage }) => {
-            console.log(errorMessage);
-            setErrorMessage(errorMessage);
-            if (errorMessage) {
+    const onConfirmPassword = useCallback(
+        (event: React.MouseEvent<HTMLButtonElement>): void => {
+            event.preventDefault();
+            updatePassword(userToEdit.userId, password).then(({ errorMessage }) => {
                 setErrorMessage(errorMessage);
-            } else {
-                onConfirm({
-                    success: true,
-                    message: (
-                        <>
-                            Password for <b>{userToEdit.email}</b> updated successfully.
-                        </>
-                    ),
-                });
-                void logInfoReturnLogId(`Password for ${userToEdit.email} updated successfully`);
-            }
-        });
-    };
+                if (errorMessage) {
+                    setErrorMessage(errorMessage);
```

Comment:
It seems like we're calling `setErrorMessage` twice in quick succession? I think we can remove the one in the conditional?